File: app/src/main/java/com/task/paginationlist/data/network/netmodels/Wallpaper.java
package com.task.paginationlist.data.network.netmodels;

import com.google.gson.annotations.SerializedName;
import com.task.paginationlist.data.db.models.WallpaperDb;


public class Wallpaper {

    @SerializedName("id")
    public long id;

    @SerializedName("bytes")
    public long size;

    @SerializedName("created_at")
    public String created;

    @SerializedName("image")
    public ImageDetails imageDetails;

    public WallpaperDb toWallpaperDb() {
        WallpaperDb result = new WallpaperDb();
        result.setServerId(id);
        result.setCreationDate(created);
        result.setImageUrl(imageDetails.preview.url);
        result.setThumbUrl(imageDetails.thumb.url);
        return result;
    }
}

File: app/src/main/java/com/task/paginationlist/data/network/netmodels/Wallpaper.kt
package com.task.paginationlist.data.network.netmodels

import com.google.gson.annotations.SerializedName


class Wallpaper {

    @SerializedName("id")
    var id: Long = 0

    @SerializedName("bytes")
    var size: Long = 0

    @SerializedName("created_at")
    var created: String? = null

    @SerializedName("image")
    var imageDetails: ImageDetails? = null

}


